oauthproxy.go

```diff
@@ -1049,3 +1072,23 @@ func (p *OAuthProxy) ErrorJSON(rw http.ResponseWriter, code int) {
 	rw.Header().Set("Content-Type", applicationJSON)
 	rw.WriteHeader(code)
 }
+
+func (p *OAuthProxy) validateGroups(groups []string) bool {
+	if len(p.AllowedGroups) == 0 {
+		return true
+	}
+
+	m := map[string]bool{}
```

Comment:
Nit: If you're using this as a set, it's conventional to use `map[string]struct{}` and then do. It is preferable to use the `ok` variable in the case below as it is explicit (and defined by the language) about whether the key exists or not
```
if _, ok := thing[key]; ok {
  return true
}
```

Reply:
Makes sense, I moved to a `map[string]struct{}` and updated the check as you mentioned.